src/pkg/control/repository/repo.go

```diff
@@ -0,0 +1,37 @@
+package repository
+
+// Repo represents options that are specific to the repo storing backed up data.
+type Options struct {
+	User string `json:"user"`
+	Host string `json:"host"`
+}
+
+type Maintenance struct {
+	Type   MaintenanceType   `json:"type"`
+	Safety MaintenanceSafety `json:"safety"`
+	Force  bool              `json:"force"`
+}
+
+// ---------------------------------------------------------------------------
+// Maintenance flags
+// ---------------------------------------------------------------------------
+
+type MaintenanceType int
+
+// Can't be reordered as we rely on iota for numbering.
+//
```

Comment:
Any reason in particular to use iota over string enums here?

Reply:
I kind of like how iotas give us the ability to have meaningful default (zero) values while also allowing string output if we use generated code. I feel that having values in a struct (for namespacing) makes it easier to have unitialized values than passing parameters to a function explicitly